src/main/mw.c

```diff
@@ -246,24 +247,20 @@ void annexCode(void)
         rcCommand[PITCH] = rcCommand_PITCH;
     }
 
-    if (feature(FEATURE_VBAT | FEATURE_CURRENT_METER)) {
-        vbatCycleTime += cycleTime;
-        if (!(++vbatTimer % VBATFREQ)) {
-
-            if (feature(FEATURE_VBAT)) {
-                updateBatteryVoltage();
-                batteryState = calculateBatteryState();
-                //handle beepers for battery levels
-                if (batteryState == BATTERY_CRITICAL)
-                    beeper(BEEPER_BAT_CRIT_LOW);    //critically low battery
-                else if (batteryState == BATTERY_WARNING)
-                    beeper(BEEPER_BAT_LOW);         //low battery
-            }
+    if (feature(FEATURE_VBAT)) {
+        /* currentTime will rollover @ 70 minutes */
+        if ((currentTime - vbatLastServiced) >= VBATINTERVAL) {
```

Comment:
This way it will call updateBattery() on each iteration for VBATINTERVAL before time wrap. Most probably it won't cause significant problem, but it may be debugging nightmare -  it takes a bit over hour to happen.

one possibility is to define
```
static inline int16_t cmp16(uint16_t a, uint16_t b) { return a-b; }
static inline int32_t cmp32(uint32_t a, uint32_t b) { return a-b; }
```

then the code is 
```
if (cmp32(currentTime, vbatLastServiced) >= VBATINTERVAL) {
vbatNextService = currentTime;
```
or
```
if (cmp32(currentTime, vbatNextService) >= 0) {
vbatNextService = currentTime + VBATINTERVAL;
```
seems perfectly readable to me ..